java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -257,6 +354,74 @@ public boolean hasNulls() {
   // DATA MOVEMENT
   /////////////////////////////////////////////////////////////////////////////
 
+  private final static HostColumnVector.NestedHostColumnVector copyToHostNestedHelper(
+      ColumnViewPointerAccess<BaseDeviceMemoryBuffer> deviceCvPointer) {
+    if (deviceCvPointer == null) {
+      return null;
+    }
+    HostMemoryBuffer hostOffsets = null;
+    HostMemoryBuffer hostValid = null;
+    HostMemoryBuffer hostData = null;
+    List<HostColumnVector.NestedHostColumnVector> children = new ArrayList<>();
+    BaseDeviceMemoryBuffer currData = null;
+    BaseDeviceMemoryBuffer currOffsets = null;
+    BaseDeviceMemoryBuffer currValidity = null;
+    long currNullCount = 0l;
+    boolean needsCleanup = true;
+    try {
+      long currRows = deviceCvPointer.getNumRows();
+      DType currType = deviceCvPointer.getDataType();
+      currData = deviceCvPointer.getDataBuffer();
+      currOffsets = deviceCvPointer.getOffsetBuffer();
+      currValidity = deviceCvPointer.getValidityBuffer();
+      if (currData != null) {
+        hostData = HostMemoryBuffer.allocate(currData.length);
+        hostData.copyFromDeviceBuffer(currData);
+      }
+      if (currValidity != null) {
+        hostValid = HostMemoryBuffer.allocate(currValidity.length);
+        hostValid.copyFromDeviceBuffer(currValidity);
+      }
+      if (currOffsets != null) {
+        hostOffsets = HostMemoryBuffer.allocate(currOffsets.length);
+        hostOffsets.copyFromDeviceBuffer(currOffsets);
+      }
+      for (int i = 0; i < deviceCvPointer.getNumChildren(); i++) {
+        ColumnViewPointerAccess childDevPtr = deviceCvPointer.getChildColumnView(i);
+        children.add(copyToHostNestedHelper(childDevPtr));
+        childDevPtr.close();
+      }
```

Comment:
Since this would eventually throw a run time exception I just closed this without a try catch. Let me know if that doesnt agree with you, @revans2 .

Reply:
@revans2 I have made some changes to the base interface and made it a try with resources block in all but one place where we are looping which makes the scopes harder. Let me know if I missed a nicer way to do this.